platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -981,6 +988,34 @@ public void onFocalPointChanged(PointF pointF) {
     }
   }
 
+  /**
+   * The initial render callback waits for rendering to happen before making the map visible for end-users.
+   * We wait for the second DID_FINISH_RENDERING_FRAME map change event as the first will still show a black surface.
+   */
+  private static class InitialRenderCallback implements OnMapChangedListener {
+
+    private WeakReference<MapView> weakReference;
+    private int renderCount;
+
+    InitialRenderCallback(MapView mapView) {
+      this.weakReference = new WeakReference<>(mapView);
+    }
+
+    @Override
+    public void onMapChanged(int change) {
+      if (change == MapView.DID_FINISH_RENDERING_FRAME) {
```

Comment:
I think there is a typo here. As you mentioned, `MapView#DID_FINISH_RENDERING_FRAME_FULLY_RENDERED` works as expected, but `MapView#DID_FINISH_RENDERING_FRAME` doesn't resolve the issue.

Reply:
forgot to document my follow up findings:
I found that `MapView#DID_FINISH_RENDERING_FRAME_FULLY_RENDERED` was taking a long time to show the map to the end user. Before core emits that event we receive multiple `MapView#DID_FINISH_RENDERING_FRAME` with every render invocation we render a bit more content overtime. First time we render the black surface, second time we render the background (this is why there is a counter there). Feel free to play around this yourself but I feel this was providing a better UX.

Reply:
Please check out the following setup in the `MapboxMap#onStart` that imitates bad connection:
```
void onStart() {
  nativeMapView.update();
  if (TextUtils.isEmpty(nativeMapView.getStyleUrl()) && TextUtils.isEmpty(nativeMapView.getStyleJson())) {
    // if user hasn't loaded a Style yet
    new Handler().postDelayed(() -> nativeMapView.setStyleUrl(Style.MAPBOX_STREETS), 2000L);
  }
}
```

Waiting for `MapView#DID_FINISH_RENDERING_FRAME_FULLY_RENDERED` here does the job well, but if you feel like it's too long, maybe you could wait for the style to load and then wait for those 2 frames to be partially rendered?

Reply:
Thank you for testing and catching that use-case! I implemented the latter suggestion and I feel it's working great for normal usage as for this specific case. LMK what you think.